client/my-sites/checkout/composite-checkout/hooks/product-variants.tsx

```diff
@@ -84,14 +84,19 @@ export function useGetProductVariants(
 			.map( ( variant ): WPCOMProductVariant | undefined => {
 				try {
 					const term = getBillingTermForMonths( variant.bill_period_in_months );
+					const introductoryTerms = variant.introductory_offer_terms;
 					return {
 						variantLabel: getTermText( term, translate ),
 						productSlug: variant.product_slug,
 						productId: variant.product_id,
 						priceInteger: variant.price_integer,
 						termIntervalInMonths: getBillingMonthsForTerm( term ),
 						termIntervalInDays: getTermDuration( term ) ?? 0,
+						introductoryInterval: introductoryTerms?.interval_count,
+						introductoryTerm: introductoryTerms?.interval_unit,
+						priceBeforeDiscounts: variant.price_before_discounts_integer,
 						currency: variant.currency,
+						planLength: variant.bill_period_in_months,
```

Comment:
Minor: I'd suggest calling this `planLengthInMonths` to make it more clear what we're measuring. Also this product might not be a plan, so maybe... `productBillingTermInMonths`?

Reply:
Naming stuff is hard, y'all :)